static/js/main.js

```diff
@@ -13245,21 +13245,58 @@ $('.js-validate-filters').each(function() {
         $gte = $this.find('.js-filter_range-date__gte'),
         $lte = $this.find('.js-filter_range-date__lte');
 
-    function validDateRange(date1, date2) {
+    // @param date1 {Date} The starting date in the range.
+    // @param date2 {Date} The ending date in the range.
+    // @return {Boolean} true if the ending date is after the starting date.
+    function isValidDateRange(date1, date2) {
         return date2.getTime() > date1.getTime();
     }
 
+    // Check that the date isn't empty,
+    // and set it to the other set date if it is.
+    // If it's still empty, return an empty string.
+    // If year or month is missing, the current month or year are added.
+    // @param targetDate {String} Date string to format.
+    // @param compData {String} Other date in the date range we're checking.
+    // @return {String} Formatted targetDate or an empty string.
+    //   Throws an error if the expected YYYY-MM format is wrong.
+    function formatDateString(targetDate, compDate) {
+        var formattedDate = targetDate === '' ? compDate : targetDate;
+        if (formattedDate === '') return formattedDate;
+
+        if (/^\d{4}$/.test(formattedDate)) {
+            var month = String(new Date().getMonth() + 1);
+            if (month.length !== 2) month = '0' + month;
+            formattedDate += '-' + month;
+        } else if (/^\d{2}$/.test(formattedDate)) {
+            var year = new Date().getFullYear();
+            formattedDate = year + '-' + formattedDate;
+        } else if (/^\d{4}-\d{2}$/.test(formattedDate) === false) {
+            var msg = 'Unexpected date format! Should be in YYYY-MM format.';
+            throw new Error(msg);
+        }
+
+        return formattedDate;
+    }
+
     $( this ).on('submit', function( e ) {
-        var validDate = validDateRange(
-            new Date( Date.parse($gte.val()) ),
-            new Date( Date.parse($lte.val()) )
+        var gteVal = $gte.val();
+        var lteVal = $lte.val();
+
+        var gteParsedVal = formatDateString(gteVal, lteVal);
+        var lteParsedVal = formatDateString(lteVal, gteVal);
+
+        var validDate = isValidDateRange(
```

Comment:
Might rename this `var validRange` for clarity on what you're testing against on [L:13293](https://github.com/cfpb/cfgov-refresh/pull/366/files#diff-93471d4bf84d0569fadcebe5fba72773R13293)